### PR TITLE
Code cleanup for micronaut-aop

### PR DESCRIPTION
**Pull request by altro3**

### CODE REVIEW

File: aop/src/main/java/io/micronaut/aop/internal/intercepted/KotlinInterceptedMethodImpl.java [hunk 1]
```diff
@@ -110,11 +111,11 @@ public CompletableFuture<Object> interceptResultAsCompletionStage() {
             );
         }
         @SuppressWarnings("unchecked")
-        CompletableFutureContinuation completableFutureContinuation = new CompletableFutureContinuation((Continuation<Object>) continuation);
+        var completableFutureContinuation = new CompletableFutureContinuation((Continuation<Object>) continuation);
         replaceContinuation.accept(completableFutureContinuation);
         Object result = context.proceed();
         replaceContinuation.accept(continuation);
-        if (result != KotlinUtils.COROUTINE_SUSPENDED) {
+        if (result != null && result != KotlinUtils.COROUTINE_SUSPENDED) {
```
Comment (dstepanov) [inline aop/src/main/java/io/micronaut/aop/internal/intercepted/KotlinInterceptedMethodImpl.java]: Why did you add a null check?

Comment (altro3) [reply aop/src/main/java/io/micronaut/aop/internal/intercepted/KotlinInterceptedMethodImpl.java]: Because `InvocationContext.proceed()` method with annotaion `@Nullable`

Comment (dstepanov) [reply aop/src/main/java/io/micronaut/aop/internal/intercepted/KotlinInterceptedMethodImpl.java]: I don't think that is related, lets see if tests are going to pass

Comment (altro3) [reply aop/src/main/java/io/micronaut/aop/internal/intercepted/KotlinInterceptedMethodImpl.java]: I can revert it

Comment (altro3) [reply aop/src/main/java/io/micronaut/aop/internal/intercepted/KotlinInterceptedMethodImpl.java]: ok, reverted

**Review by graemerocher — APPROVED** [review]

**Review by dstepanov — APPROVED** [review]
